environment/src/main/java/com/sequenceiq/environment/parameters/validation/validators/parameter/AzureParameterValidator.java

```diff
@@ -124,11 +127,21 @@ private ValidationResult validateEntitlement(ValidationResultBuilder validationR
                                         + "but that feature is currently disabled").build();
                     }
                 default:
-                    return validationResultBuilder.build();
+                    break;
+            }
+        }
+
+        String encryptionKeyUrl = azureResourceEncryptionParametersDto.getEncryptionKeyUrl();
+        if (Objects.nonNull(encryptionKeyUrl)) {
+            if (!entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)) {
+                LOGGER.info("Invalid request, CDP_CB_AZURE_DISK_SSE_WITH_CMK entitlement turned off for account {}", accountId);
+                return validationResultBuilder.error(
+                        "You specified encryptionKeyUrl to use Server Side Encryption for Azure Managed disks with CMK, "
+                                + "but that feature is currently disabled. Enable 'CDP_CB_AZURE_DISK_SSE_WITH_CMK' to use SSE with CMK.").
```

Comment:
I would prefer a more user-friendly error message here, that emphasise that the Server Side Encryption related entitlement is not enabled for your account and please contact with us. I think `CDP_CB_AZURE_DISK_SSE_WITH_CMK` means nothing for a customer as they couldn't set entitlement explicitly AFAIK.